Sources/TurnkeyStamper/Internal/SecureStorageStamper.swift

```
@@ -0,0 +1,407 @@
+    externalKeyPair: (publicKey: String, privateKey: String)? = nil
+  ) throws -> String {
+    let privateKeyHex: String
+    let publicKeyHex: String
+
+    if let provided = externalKeyPair {
+      privateKeyHex = provided.privateKey
+      publicKeyHex = provided.publicKey
+    } else {
+      let priv = P256.Signing.PrivateKey()
+      privateKeyHex = priv.rawRepresentation.toHexString()
+      publicKeyHex = priv.publicKey.compressedRepresentation.toHexString()
+    }
+
+    try savePrivateKey(privateKeyHex, for: publicKeyHex, config: SecureStorageConfig())
+    return publicKeyHex
+  }
+
+  static func createKeyPair(config: SecureStorageConfig) throws -> String {
+    let priv = P256.Signing.PrivateKey()
+    let privateKeyHex = priv.rawRepresentation.toHexString()
+    let publicKeyHex = priv.publicKey.compressedRepresentation.toHexString()
+    try savePrivateKey(privateKeyHex, for: publicKeyHex, config: config)
+    return publicKeyHex
+  }
+
+  static func createKeyPair(
+    externalKeyPair: (publicKey: String, privateKey: String),
+    config: SecureStorageConfig
+  ) throws -> String {
+    try savePrivateKey(externalKeyPair.privateKey, for: externalKeyPair.publicKey, config: config)
+    return externalKeyPair.publicKey
+  }
+
+  static func deleteKeyPair(publicKeyHex: String) throws {
+    var query: [String: Any] = [
+      kSecClass as String: kSecClassGenericPassword,
+      kSecAttrService as String: publicKeyHex,
+      kSecAttrAccount as String: account,
+    ]
+
+    let status = SecItemDelete(query as CFDictionary)
+    if status == errSecItemNotFound || status == errSecSuccess {
+      return
+    }
+    throw SecureStorageStamperError.keychainError(status)
+  }
+
+  /// Delete a specific key using the provided config to match how it was stored (e.g., access group,
+  /// iCloud Keychain). Use when you stored with non-default attributes.
+  static func deleteKeyPair(publicKeyHex: String, config: SecureStorageConfig) throws {
```

lets add a comment here that we also silently ignore if the key isn't found here!